openscvx/plotting/plotting.py

```diff
@@ -1,279 +1,298 @@
-import numpy as np
 import plotly.graph_objects as go
 from plotly.subplots import make_subplots
 
 from openscvx.algorithms import OptimizationResults
-from openscvx.config import Config
 
 
-def plot_state(
-    result: OptimizationResults = None,
-    params: Config = None,
-    problem=None,
-    state_names=None,
+def _get_var_dim(result: OptimizationResults, var_name: str, var_list: list) -> int:
+    """Get dimensionality of a variable from the metadata."""
+    for var in var_list:
+        if var.name == var_name:
+            s = var._slice
+            if isinstance(s, slice):
+                return (s.stop or 1) - (s.start or 0)
+            return 1
+    raise ValueError(f"Variable '{var_name}' not found")
+
+
+def _plot_timeseries(
+    fig: go.Figure,
+    result: OptimizationResults,
+    var_name: str,
+    component_idx: int,
+    row: int,
+    col: int,
+    show_legend: bool,
+    show: str = "both",
 ):
-    """Plot state trajectories over time with bounds.
+    """Add traces for a single component of a variable to the figure."""
+    t_nodes = result.nodes["time"].flatten()
+    has_trajectory = bool(result.trajectory) and var_name in result.trajectory
+    t_full = result.trajectory["time"].flatten() if has_trajectory else None
+
+    # Plot propagated trajectory if available
+    if show in ("both", "trajectory") and has_trajectory:
+        data = result.trajectory[var_name]
+        y = data if data.ndim == 1 else data[:, component_idx]
+        fig.add_trace(
+            go.Scatter(
+                x=t_full,
+                y=y,
+                mode="lines",
+                name="Trajectory",
+                showlegend=show_legend,
+                legendgroup="trajectory",
+                line={"color": "green", "width": 2},
+            ),
+            row=row,
+            col=col,
+        )
+
+    # Plot optimization nodes
+    if show in ("both", "nodes") and var_name in result.nodes:
+        data = result.nodes[var_name]
+        y = data if data.ndim == 1 else data[:, component_idx]
+        fig.add_trace(
+            go.Scatter(
+                x=t_nodes,
+                y=y,
+                mode="markers",
+                name="Nodes",
+                showlegend=show_legend,
+                legendgroup="nodes",
+                marker={"color": "cyan", "size": 6, "symbol": "circle"},
+            ),
+            row=row,
+            col=col,
+        )
 
-    Shows the optimized state trajectory (nodes and full propagation if available),
-    initial guess, and constraint bounds for all state variables.
+
+def plot_state(
+    result: OptimizationResults,
+    state_name: str,
+    show: str = "both",
```

Comment:
I am in favor of just removing this entirely as an argument and using a legendgroup to group all the nodal and propagation traces together. Then the user can use the legend to toggle the traces off or on.